### ISSUE TITLE
Use Transient menus?

### ISSUE THREAD

**Issue by alphapapa** [jdtsmith/speedrect#6]
Hi, this package looks very useful.  Have you considered using Transient for the menus?  I think it would be like the icing on the cake.  :)

**Comment by jdtsmith** [comment]
Thanks.  RE transient, will think on it.  In general I like to do transients if they are more than an attractive shortcut list.  Would you say keep the transient up all the time until quit?  Some may find that intrusive.  One other wrinkle: at least one of the rectangle functions in emacs (`string-rectangle`) already does its own prompting with "live update".

**Comment by alphapapa** [comment]
> Would you say keep the transient up all the time until quit? Some may find that intrusive.

Yes, that would absolutely be helpful, and in fact the latest release of Transient has improvements for that workflow.  I'm pretty sure it could also have a user option to disable that, but I think most users would keep it up (I know I would).

> One other wrinkle: at least one of the rectangle functions in emacs (string-rectangle) already does its own prompting with "live update".

That one could probably be made an exception.  Transient is very flexible in this regard.

**Comment by jdtsmith** [comment]
BTW, I'd be open to a PR for this.  I still find transient rather demoralizing to work with.  Which is a shame because it's such a cool interface.  

With the new "restart speedrect" functionality for most quick-commands, I am using it far more often.  It's super convenient and I reach for it more than multiple-cursors now.

**Comment by alphapapa** [comment]
> BTW, I'd be open to a PR for this. I still find transient rather demoralizing to work with. Which is a shame because it's such a cool interface.

I don't use this package yet, so I won't be preparing a PR.  Anyway, Transient isn't bad to work with for simple things.  This repo also has examples of more advanced usage: https://github.com/positron-solutions/transient-showcase

**Comment by jdtsmith** [comment]
No problem, happy to revisit if that changes. 

Yes I'm familiar with the showcase (was in fact one of the ones advocating for it). It's _still_ hard to just write out an interface. The API is just so flexible but somewhat uneven in its constructs (though getting better).

**Comment by stottm** [comment]
Reach out to Charles Choi - Kickingvegas/Casual he's been building transient menus for many things. He just had an EmacsConf video 2024.  Typically binds C-o to open the transient. I think a video demo would go a long way in demonstrating the usefulness of speedrect.

**Comment by jdtsmith** [comment]
> I think a video demo would go a long way

New video demo just added.
